### PR TITLE
chore: replace dependency on jx-api

### PR DESCRIPTION
**Pull request by garethjevans**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

**Comment by abayer** [comment]
Codegen needs to get reworked some - failing on generating CRD ref docs currently.

**Comment by garethjevans** [comment]
Yeah, I've not tackled the codegen stuff yet, am more concerned about some other parts of the generation.  It looks like the `DeepCopy` stuff has been generated slightly different, e.g.

```
3575c3575
< 	in.ListMeta.DeepCopyInto(&out.ListMeta)
---
> 	out.ListMeta = in.ListMeta
```

**Comment by hferentschik** [comment]
`jx-api` only encompasses the Jenkins X CRDs atm, right?
Probably worth creating an issue in `jx` for the switch.

**Comment by garethjevans** [comment]
@hferentschik its `pkg/apis`,`pkg/clients` & some of `pkg/config` to allow the install requirements to work.

File: Makefile [hunk 1]
```diff
@@ -220,4 +220,3 @@ code-generate:
 	./hack/generate.sh
 
 include Makefile.docker
-include Makefile.codegen
```
Comment (abayer) [inline Makefile:223]: So we do still need/want codegen for the deepcopy structs for `pkg/tekton/syntax`, `pkg/jenkinsfile`, and `pkg/config` (for `ProjectConfig` in the last case).

File: Makefile.codegen [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# Anything generated by this target should be checked in
+generate: build-codegen install-generate-deps generate-mocks generate-client ## Generate the Go code (crds, mocks, openapi, client)
+	@$(MAKE) fmt
+	@$(MAKE) importfmt
+	@echo "Generation complete"
+
+install-generate-deps:
+	$(GO) get github.com/petergtz/pegomock/...@$(PEGOMOCK_VERSION)
+
+generate-mocks: install-generate-deps go-generate fmt importfmt ## Generate the mocks
```
Comment (abayer) [inline Makefile.codegen]: Oh yeaaaaaah - I forgot about the mocks! Good catch.

**Comment by abayer** [comment]
/override tf-boot

It actually passed everything but the known destroy issue.

**Pull request merged**